catalog/core/catalog-core-api-impl/src/test/java/ddf/catalog/impl/MockSource.java

```diff
@@ -109,7 +111,11 @@ public String getVersion() {
 
     @Override
     public SourceResponse query(QueryRequest query) {
-        return null;
+        return sourceResponse;
+    }
+
+    public void setQueryResponse(SourceResponse sourceResponse) {
```

Comment:
Yes - originally `query()` in the `MockSource` class simply returned `null`. I needed it to be able to return a meaningful value for my unit tests. It still returns `null` as a default value, I only added the option to change it.